From 3a0f81eb9d7a2667d0856a43575eae968e223329 Mon Sep 17 00:00:00 2001
From: svenate <salini.venate@intel.com>
Date: Wed, 16 Mar 2022 13:46:58 +0530
Subject: [PATCH] WA-Move -pie option from compiler to linker flags

Move -pie option from compiler to linker flags to
fix below build error with clang 14.02:
clang++: error: argument unused during compilation:
'-pie' [-Werror,-Wunused-command-line-argument]

Signed-off-by: svenate <salini.venate@intel.com>
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 mfx_omx_defs.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mfx_omx_defs.mk b/mfx_omx_defs.mk
index ebffab1..cdeeb9d 100644
--- a/mfx_omx_defs.mk
+++ b/mfx_omx_defs.mk
@@ -82,7 +82,7 @@ MFX_OMX_CFLAGS += -Wall -Werror
 #  Security
 MFX_OMX_CFLAGS += \
   -fstack-protector \
-  -fPIE -fPIC -pie \
+  -fPIE -fPIC \
   -O2 -D_FORTIFY_SOURCE=2 \
   -Wformat -Wformat-security \
   -fexceptions -frtti
@@ -128,7 +128,7 @@ endif
 
 # Setting usual link flags
 MFX_OMX_LDFLAGS := \
-  -z noexecstack \
+  -z noexecstack -pie \
   -z relro -z now
 
 # Setting vendor
-- 
2.17.1

